### DISCUSSION TITLE
Padrão de exibição de mensagens de API

### DISCUSSION THREAD

**Discussion by caiosantosf** [po-ui/po-angular#998] [Tire suas dúvidas]
Boa tarde, tenho uma situação onde minha API retorna duas mensagens de erro num response com status 400.

O JSON retornado da API para este exemplo é:

![image](https://user-images.githubusercontent.com/14207713/132568196-86044a1e-6436-45d8-9895-8edb5eab4cf9.png)
imagem 1

No web-app é exibido a notificação da seguinte forma:

![image](https://user-images.githubusercontent.com/14207713/132568325-44003749-b7f0-4948-8669-aa374e683238.png)
 imagem 2

Clicando em detalhes exibe as minhas duas mensagens mas também repete o texto da mensagem "cabeçalho", o que consideramos meio confuso para o usuário.

![image](https://user-images.githubusercontent.com/14207713/132568468-614a6f63-1a90-4158-8eb5-8ee8b539318f.png)
imagem 3

Lendo a documentação do PO-UI pude ver que para responses com status 200 é possível passar um array de mensagens dentro de uma propriedade "_messages", quando isso é feito o PO-UI me mostra duas notificações direto na tela, sem precisar acessar detalhes, o que na nossa visão fica melhor para o usuário.

![image](https://user-images.githubusercontent.com/14207713/132570552-4d249bc3-0ada-4eff-b35a-6af5f5f1075d.png)
imagem 4

![image](https://user-images.githubusercontent.com/14207713/132571309-8051b263-0a22-4b87-8480-9e95139f00b0.png)
imagem 5

Porém esta alternativa com o array _messages não funciona com o status 400 e nesse nosso caso retornar status 200 seria muito inadequado pois quebra a semântica HTTP da API.

Gostaria de ver com vocês como veem esse caso para padronizarmos o retorno de múltiplas mensagens. Seria possível implementar no PO-UI para que quando houver status 400 tambem poder retornar o array _messages, ou, alterar para que quando as mensagens ficarem agrupadas nos detalhes não exibir neles o texto da notificação (o "Ocorreram erros de validação. Clique em Detalhes para ver" vide a terceira imagem do post) ?

Desde já deixo o meu obrigado!

**Comment by jhonyeduardo** [comment]
Bom dia @caiosantosf 

Implementamos a propriedade "detailTitle", onde você pode colocar outra mensagem, isso resolveria seu problema, pois
a "message" seria exibida na Notificação e o "detailTitle" no accordion da modal.

**Reply by jhonyeduardo** [reply]
https://po-ui.io/documentation/po-http-interceptor

**Comment by caiosantosf** [comment]
Obrigado @jhonyeduardo! Essa propriedade realmente melhora nossa usabilidade com as notificações.
